packages/compass-sidebar/src/components/legacy/sidebar-databases-navigation.tsx

```diff
@@ -80,46 +86,74 @@ function SidebarDatabasesNavigation({
 
   return (
     <ConnectionsNavigationTree
+      connections={connections}
       {...dbNavigationProps}
-      databases={databases}
       onNamespaceAction={onNamespaceAction}
       isReadOnly={isReadOnly}
     />
   );
 }
 
-function mapStateToProps(state: RootState) {
+function mapStateToProps(
+  state: RootState,
+  { connectionInfo }: { connectionInfo: ConnectionInfo }
+): {
+  isReady: boolean;
+  connections: Connection[];
+  expanded: Record<string, Record<string, boolean> | false>;
+} {
+  const connectionId = connectionInfo.id;
+  const instance = state.instance[connectionId];
   const {
-    databases: { filterRegex, filteredDatabases, expandedDbList },
-    instance,
-  } = state;
+    filterRegex,
+    filteredDatabases,
+    expandedDbList: initialExpandedDbList,
+  } = state.databases[connectionId] || {};
+
   const status = instance?.databasesStatus;
   const isReady =
     status !== undefined && !['initial', 'fetching'].includes(status);
   const defaultExpanded = Boolean(filterRegex);
+
+  const expandedDbList = initialExpandedDbList ?? {};
   const expanded = Object.fromEntries(
-    (filteredDatabases as any[]).map(({ name }) => [
+    ((filteredDatabases as any[]) || []).map(({ name }) => [
       name,
       expandedDbList[name] ?? defaultExpanded,
     ])
   );
-  const isDataLake = instance?.dataLake.isDataLake;
-  const isWritable = instance?.isWritable;
+
+  const isDataLake = instance?.dataLake?.isDataLake ?? false;
+  const isWritable = instance?.isWritable ?? false;
 
   return {
-    isReady,
-    isDataLake,
-    isWritable,
-    databases: filteredDatabases,
-    expanded,
+    isReady: true,
+    connections: [
+      {
+        isReady,
+        isDataLake,
+        isWritable,
+        name: '',
+        connectionInfo,
+        databasesLength: filteredDatabases?.length || 0,
+        databasesStatus: (status ??
+          'fetching') as Connection['databasesStatus'],
+        databases: filteredDatabases ?? [],
+      },
+    ],
+    expanded: {
+      [connectionId]: expanded,
+    },
```

Comment:
We probably should do this inside of the component with a memo. This will pass new connections and expanded as props for every state update in redux which will make the underlying component re-render everytime.
(We can surely do a follow up for this - to avoid holding this PR for too long)

Reply:
Maybe we can pair on this one, because I'm not sure how to move it to useMemo as it depends on the redux state.